### PR TITLE
Added serviceUpStream annotations

### PR DESCRIPTION
**Pull request by manishjaiswal2**

### CODE REVIEW

File: src/common/templates/_ingress.tpl [hunk 1]
```diff
@@ -26,6 +26,9 @@ metadata:
     {{- if $.Values.global.ingress.objects.annotations }}
     {{- include "harnesscommon.tplvalues.render" (dict "value" $.Values.global.ingress.objects.annotations "context" $) | nindent 4 }}
     {{- end }}
+    {{- if $.Values.ingress.serviceUpStream.enabled }}
```
Comment (arya-harness) [inline src/common/templates/_ingress.tpl:29]: hmm, not like this. make it controllable via values.yaml 
when you provide an annotation you can wrap it under a condition.
also bump up the chart version in Chart.yaml

File: src/common/templates/_ingress.tpl [hunk 2]
```diff
@@ -26,6 +26,9 @@ metadata:
     {{- if $.Values.global.ingress.objects.annotations }}
     {{- include "harnesscommon.tplvalues.render" (dict "value" $.Values.global.ingress.objects.annotations "context" $) | nindent 4 }}
     {{- end }}
+    {{- if $.Values.ingress.serviceUpStream.annotations }}
```
Comment (arya-harness) [inline src/common/templates/_ingress.tpl:29]: Not recommended like this:
add a field under ingress objects which are under conditions. like conditionalAnnotations
which expects an array of annotations and conditions associated with it

```
ingress:
  objects:
    - annotations: 
        nginx.ingress.kubernetes.io/rewrite-target: /$2
       conditionalAnnotations: <-- example
       - newAnnotation: newValue
          condition: "global.upstream.enabled"
      paths:
      - path: '{{ .Values.global.ingress.pathPrefix }}/pipeline(/|$)(.*)'

 ```

Comment (manishjaiswal2) [reply src/common/templates/_ingress.tpl]: made changes as per your request. please validate.

**Review by arya-harness — APPROVED** [review]
syntax
```
- name: harness-manager-api
      annotations: 
        nginx.ingress.kubernetes.io/proxy-read-timeout: "300"
        nginx.ingress.kubernetes.io/rewrite-target: /api$1$2
      conditionalAnnotations:
      - condition: "global.ccm.enabled"
        annotations:
          abc: def
 ```

**Pull request merged**